samcli/commands/docs/command.py

```diff
@@ -13,27 +21,114 @@
     AWS SAM CLI lifecycle and other useful details. 
 """
 
+SUCCESS_MESSAGE = "Documentation page opened in a browser"
+ERROR_MESSAGE = "Failed to open a web browser. Use the following link to navigate to the documentation page: {URL}"
+
+
+class InvalidDocsCommandException(UserException):
+    """
+    Exception when the docs command fails
+    """
+
+
+class DocsBaseCommand(click.Command):
+    def __init__(self, *args, **kwargs):
+        command_callback = DocsCommand().command_callback
+        super().__init__(name="docs", callback=command_callback)
+
+
+class DocsSubcommand(click.MultiCommand):
+    def __init__(self, command: Optional[List[str]] = None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.docs_command = DocsCommand()
+        self.command = command or self.docs_command.sub_commands
+        self.command_string = self.docs_command.sub_command_string
+        self.command_callback = self.docs_command.command_callback
+
+    def get_command(self, ctx: Context, cmd_name: str) -> Command:
```

Comment:
how does --help text look like?

Reply:
I used the new formatters to format the help text. Looks like this for the base command and all sub-commands.

![image](https://user-images.githubusercontent.com/84205762/226469831-1fd14e58-d99c-430b-a6d5-2c043f317ad2.png)